### ISSUE TITLE
DL3009 and DL3060 false negative for Dockerfile with multiple targets

### ISSUE THREAD

**Issue by miwig** [hadolint/hadolint#969]
- [x] This is a bug report
- [x] I searched existing issues before opening this one

A Dockerfile can be used to [build multiple images/targets.](https://docs.docker.com/build/guide/multi-stage/#build-targets)

### Expected behavior
The existence of a second target should not influence warnings for other targets
### Actual behavior
Adding a second target silences some warnings for the first target
### Steps to reproduce the behavior

Run hadolint on a Dockerfile containing multiple targets

**Output of `hadolint --version`:**

```bash
Haskell Dockerfile Linter 2.12.0
```

**Dockerfile**

```Dockerfile
FROM debian:bookworm AS image1
# no warning:
RUN apt-get update -y && apt-get -y upgrade -y # DL3009
RUN yarn install # DL3060
# warning correctly:
RUN yum install -y httpd-2.24.2
RUN zypper install -y httpd=2.4.46
RUN dnf install -y httpd-2.24.2

# all rules work if this line is removed:
FROM alpine:3 AS image2
```

**Hadolint output:**
```
Dockerfile:6 DL3032 warning: `yum clean all` missing after yum command.
Dockerfile:7 DL3036 warning: `zypper clean` missing after zypper use.
Dockerfile:8 DL3040 warning: `dnf clean all` missing after dnf command.
```

**Expected output:**
```
Dockerfile:3 DL3009 info: Delete the apt-get lists after installing something
Dockerfile:4 DL3060 info: `yarn cache clean` missing after `yarn install` was run.
Dockerfile:6 DL3032 warning: `yum clean all` missing after yum command.
Dockerfile:7 DL3036 warning: `zypper clean` missing after zypper use.
Dockerfile:8 DL3040 warning: `dnf clean all` missing after dnf command.
```
